### PR TITLE
Add metric for time taken to sync Gerrit changes

### PR DESCRIPTION
**Pull request by timwangmusic**
@listx @cjwagner @mpherman2

### CODE REVIEW

File: prow/gerrit/adapter/adapter.go
Comment (chaodaiG) [inline prow/gerrit/adapter/adapter.go:241]: why not reuse `now`?

Comment (cjwagner) [reply prow/gerrit/adapter/adapter.go:241]: We need a timestamp *after* the query completes to measure the time it takes for us to process the changes.

**Review by cjwagner — APPROVED** [review]
Its a bit unfortunate that we're mixing prometheus metrics with log based metrics (which take an extra, manual step to configure), but we we're already doing that here so its ok with me if you want to keep this. In my opinion, ideally we'd align on one with my preference being prometheus metrics since those are defined entirely in code and don't require log based metric rules.
/hold

File: prow/gerrit/adapter/adapter.go
Comment (listx) [inline prow/gerrit/adapter/adapter.go:265]: @timwangmusic Can you try converting this to a Prometheus metric instead? I assume it would be straightforward. Plus it's OK if the exact parameters aren't correct in your first attempt because at least everything is explicit and Cole and review it before it's merged.

Comment (timwangmusic) [reply prow/gerrit/adapter/adapter.go]: Done.

File: prow/gerrit/adapter/adapter.go
Comment (listx) [inline prow/gerrit/adapter/adapter.go:248]: Nit: rename to `timeQueryingForChanges` or maybe even `timeQueryChangesForInstance` to match the function call name.

Comment (timwangmusic) [reply prow/gerrit/adapter/adapter.go]: Done.

**Review by cjwagner — APPROVED** [review]

**Pull request merged**
